### DISCUSSION TITLE
Firebase AppCheck in SSR - Error

### DISCUSSION THREAD

**Discussion by rcheiko** [nuxt/nuxt#18620] [Questions]
Hello, 
I'm trying to add AppCheck in my website on SSR in api/lib/firebase.ts

```
const config = useRuntimeConfig();
const firebaseConfig = {
  apiKey: config.firebaseApiKey,
  authDomain: "xxxxxx",
  projectId: "xxxxxx",
  storageBucket: "xxxxxx",
  messagingSenderId: "xxxxxx",
  appId: "xxxxxx",
  measurementId: "xxxxxx",
};

const app = initializeApp(firebaseConfig);

// if (process.client) {
  console.log("Initializing Firebase App Check");
  initializeAppCheck(app, {
    provider: new ReCaptchaEnterpriseProvider(config.public.recaptchaApiKey),
    isTokenAutoRefreshEnabled: true, // Set to true to allow auto-refresh.
  });
// }

const db = getFirestore(app);

export { db };
```

When i'm calling the initializeAppCheck i have a document defined error ( in website it render that : Cannot access 'renderer$1' before initialization )
I don't know how to solve the error or where to call it, i'm struggle for like 2 days if someone have an idea, thank you

**Comment by Baroshem** [comment]
It seems that this initializeAppCheck method needs to be called on the client side to have access to the document.

Have you tried registering it onMounted or in client only plugin?
